File: SORT-CHAR-BY-FREQUENCY/scbf.java
class Solution {
    public String frequencySort(String s) {
        Map<Character, Integer> hm = new HashMap<>();

        for (char p : s.toCharArray()) {
            hm.put(p, hm.getOrDefault(p, 0) + 1);
        }

        List<Map.Entry<Character, Integer>> list = new ArrayList<>(hm.entrySet());

        Collections.sort(list, (a, b) -> b.getValue() - a.getValue());

        StringBuilder result = new StringBuilder();
        for (Map.Entry<Character, Integer> entry : list) {
            char ch = entry.getKey();
            int freq = entry.getValue();
            for (int i = 0; i < freq; i++) {
                result.append(ch);
            }
        }

        return result.toString();
    }
}

File: SORT-CHAR-BY-FREQUENCY/scbf.cpp
class Solution {
public:
    string frequencySort(string s) {
        unordered_map<char, int> ht;

        for (char c : s) {
            ht[c]++;
        }

        vector<pair<char, int>> vec(ht.begin(), ht.end());

        sort(vec.begin(), vec.end(), [](const pair<char, int>& a, const pair<char, int>& b) {
            return a.second > b.second;
        });

        string result;
        for (auto& p : vec) {
            result.append(p.second, p.first);
        }

        return result;
    }
};